server/src/main/java/io/druid/server/QueryResource.java

```diff
@@ -257,7 +261,7 @@ public void write(OutputStream outputStream) throws IOException, WebApplicationE
 
                     os.flush(); // Some types of OutputStream suppress flush errors in the .close() method.
                     os.close();
-
+                    successfulQueryCount++;
```

Comment:
Is `++` guaranteed atomic? My understanding is that this is not guaranteed to atomic across multiple threads calling it at the same time. Unless you can tell for sure that `++` is java guaranteed atomic on a volatile, then it would be safer to use an `AtomicLong`